mlptrain/sampling/plumed.py

```diff
@@ -1033,6 +1033,43 @@ def __init__(self, name: str, atom_groups: Sequence = None):
         )
 
 
+class PlumedCNCV(_PlumedCV):
+    def __init__(self, name: str, ref: float, atom_groups: Sequence = None):
+        """
+        PLUMED collective variable as a coordinattion number,
+
+        e.g. [(0, 2)] gives ζ =[1-(r_02/r_ref)^6]/ [1-(r_02/r_ref)^12]
+
+        -----------------------------------------------------------------------
+        Arguments:
+
+            name: (str) Name of the collective variable
+
+            ref: (float) Reference values to distinguish different states
+
+            atom_groups: (Sequence[Sequence[int]]) List of atom index sequences
+                                                which are used to generate DOFs
+        """
+
+        super().__init__(name=name, atom_groups=atom_groups)
+
+        self._set_units()
+
+        self.ref = ref
+
+        func = f'(1-({self.dof_names[0]}/{self.ref})^6)/(1-({self.dof_names[0]}/{self.ref})^12)'
```

Comment:
Currently, the CN has hardcoded parameters of the switching function, i.e., 6 and 12, could this be changed into variables to allow for more flexibility? For example, the metal complexes often need to use different shapes of the switching functions.

Reply:
added